code/controllers/Processes/explosives.dm

```diff
@@ -0,0 +1,200 @@
+var/datum/controller/process/explosives/bomb_processor
+
+// yes, let's move the laggiest part of the game to a process
+// nothing could go wrong -- Lohikar
+/datum/controller/process/explosives
+	var/list/work_queue
+	var/ticks_without_work = 0
+	var/powernet_rebuild_was_deferred_already
+
+/datum/controller/process/explosives/setup()
+	name = "explosives"
+	schedule_interval = 5 // every half-second
+	work_queue = list()
+	bomb_processor = src
+
+/datum/controller/process/explosives/doWork()
+	if (!work_queue)
+		setup()	// fix for process failing to re-init after termination
+
+	if (!(work_queue.len))
+		ticks_without_work++
+		if (ticks_without_work > 5)
+			if(defer_powernet_rebuild)
+				makepowernets()
+				defer_powernet_rebuild = 0
+		return
+
+	ticks_without_work = 0
+
+	for (var/datum/explosiondata/data in work_queue)
+		SCHECK
+		var/turf/epicenter = data.epicenter
+		var/devastation_range = data.devastation_range
+		var/heavy_impact_range = data.heavy_impact_range
+		var/light_impact_range = data.light_impact_range
+		var/flash_range = data.flash_range
+		var/adminlog = data.adminlog
+		var/z_transfer = data.z_transfer
+
+		var/power = max(0,devastation_range) * 2 + max(0,heavy_impact_range) + max(0,light_impact_range)
+		if(config.use_recursive_explosions)
+			explosion_rec(epicenter, power)
```

Comment:
This is still unprotected from lag. Should also do that, because some procs go straight to this.